File: src/main/java/org/seasar/doma/it/dao/ScriptDao.java
package org.seasar.doma.it.dao;

import org.seasar.doma.AccessLevel;
import org.seasar.doma.Dao;
import org.seasar.doma.Script;
import org.seasar.doma.jdbc.Config;

@Dao(accessLevel = AccessLevel.PACKAGE)
public interface ScriptDao {

  static ScriptDao get(Config config) {
    return new ScriptDaoImpl(config);
  }

  @Script
  void create();

  @Script(haltOnError = false)
  void drop();
}

File: kotlin/src/main/kotlin/org/seasar/doma/it/kotlin/dao/ScriptDao.kt
package org.seasar.doma.it.kotlin.dao

import org.seasar.doma.Dao
import org.seasar.doma.Script
import org.seasar.doma.jdbc.SqlLogType

@Dao
interface ScriptDao {
    @Script(sqlLog = SqlLogType.NONE)
    fun create()

    @Script(haltOnError = false, sqlLog = SqlLogType.NONE)
    fun drop()
}
